easybuild/tools/run.py

```diff
@@ -425,12 +425,20 @@ def to_cmd_str(cmd):
             # -1 means reading until EOF
             read_size = 128 if exit_code is None else -1
 
-            more_stdout = proc.stdout.read1(read_size) or b''
-            stdout += more_stdout
+            # get output as long as output is available;
+            # note: can't use proc.stdout.read without read_size argument,
+            # since that will always wait until EOF
+            more_stdout = True
+            while more_stdout:
+                more_stdout = proc.stdout.read(read_size) or b''
+                stdout += more_stdout
 
             # note: we assume that there won't be any questions in stderr output
             if split_stderr:
-                stderr += proc.stderr.read1(read_size) or b''
+                more_stderr = True
+                while more_stdout:
```

Comment:
```suggestion
                while more_stderr:
```

Reply:
fixed in c090f49056eccf165e3a493c9ad0c8b19d210817